File: java/src/main/java/org/sammancoaching/parameterizedtests/Status.java
package org.sammancoaching.parameterizedtests;

/**
 * Enumeration for all the statuses an {@link Order} can be in.
 */
public enum Status {

	/**
	 * Placed, but not payed yet. Still changeable.
	 */
	PAYMENT_EXPECTED,

	/**
	 * {@link Order} was payed. No changes allowed to it anymore.
	 */
	PAID,

	/**
	 * The {@link Order} is currently processed.
	 */
	PREPARING,

	/**
	 * The {@link Order} is ready to be picked up by the customer.
	 */
	READY,

	/**
	 * The {@link Order} was completed.
	 */
	TAKEN;
}

File: typescript/src/status.ts
/**
 * Enumeration for all the statuses an Order can be in.
 */
export enum Status {
  /**
   * Placed, but not paid yet. Still changeable.
   */
  PAYMENT_EXPECTED = "PAYMENT_EXPECTED",

  /**
   * Order was paid. No changes allowed to it anymore.
   */
  PAID = "PAID",

  /**
   * The Order is currently processed.
   */
  PREPARING = "PREPARING",

  /**
   * The Order is ready to be picked up by the customer.
   */
  READY = "READY",

  /**
   * The Order was completed.
   */
  TAKEN = "TAKEN",
}